pkg/job_controller/pod.go

```diff
@@ -346,6 +346,10 @@ func (jc *JobController) createNewPod(job interface{}, rt, index string, spec *a
 
 	// Set name for the template.
 	podTemplate.Name = GenGeneralName(metaObject.GetName(), rt, index)
+	// Compatible with the naming convention of ElasticDL
+	if jc.Controller.ControllerName() == "ElasticDLController" && masterRole {
+		podTemplate.Name = "elasticdl-" + metaObject.GetName() + "-master"
```

Comment:
In kubedl, the master pod name is `master-jobname-0`. In elasticdl, the master pod name is `elasticdl-jobname-master`.

Please refer to https://github.com/sql-machine-learning/elasticdl/blob/develop/elasticdl_client/common/k8s_client.py#L33